core/src/main/java/org/kiwix/kiwixmobile/core/search/viewmodel/effects/SearchInPreviousScreen.kt

```diff
@@ -18,22 +18,15 @@
 
 package org.kiwix.kiwixmobile.core.search.viewmodel.effects
 
-import android.app.Activity
-import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import org.kiwix.kiwixmobile.core.base.SideEffect
-import org.kiwix.kiwixmobile.core.utils.TAG_FILE_SEARCHED
+import org.kiwix.kiwixmobile.core.extensions.ActivityExtensions.popNavigationBackstack
+import org.kiwix.kiwixmobile.core.main.CoreMainActivity
 
 data class SearchInPreviousScreen(private val searchString: String) : SideEffect<Unit> {
   override fun invokeWith(activity: AppCompatActivity) {
-    activity.setResult(
-      Activity.RESULT_OK,
-      Intent().apply {
-        putExtra(EXTRA_SEARCH_IN_TEXT, true)
-        putExtra(TAG_FILE_SEARCHED, searchString)
-      }
-    )
-    activity.finish()
+    activity.popNavigationBackstack()
+    (activity as CoreMainActivity).findInPage(searchString)
```

Comment:
https://stackoverflow.com/a/60757744/12225251 not sure if this is much of an option?

Reply:
hmm, will check out.

Reply:
It works very nicely with one exception. The fragment result is not cleared on observation. This means that if the user first searches inside a page, and then opens a page, the search in page will open as well. I solved it by manually clearing the observable ID after observation, but I am suspecting this is not the correct way to do it. Thoughts? 

```
  fun <T> Activity.observeNavigationResult(
    key: String,
    owner: LifecycleOwner,
    observer: Observer<T>
  ) {
    getObservableNavigationResult<T>(key)?.observe(owner) {
      observer.onChanged(it)
      asCoreMainActivity().consumeObservable<T>(key) <------------------
    }
  }
  fun <T> Activity.consumeObservable(key: String = "result") =
    asCoreMainActivity().navController.currentBackStackEntry?.savedStateHandle?.remove<T>(key)

```

Reply:
```kotlin
    requireActivity().observeNavigationResult<SearchItemToOpen>(
      TAG_FILE_SEARCHED,
      viewLifecycleOwner,
      Observer(::openSearchItem)
    )
```

Reply:
You might need to present a more bug style report for me to understand this quickly

Reply:
Good idea!

**Describe the bug**
<!-- A clear and concise description of what the bug is. -->
Navigation results remain in `savedStateHandle`, the symptom of this is that on entry to readerFragment, the search bar will always open. 

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->
Once a navigation result has been read, it should be removed from the `savedStateHandle`. 

**Steps to reproduce the behavior:**
1. Enter search from the reader fragment.
2. Type "t" and click "Find in page".
3. Notice that the search bar opens in the reader fragment, and that you can navigate between found words. This is as expected.
4. Close the search bar by pressing the back button.
5. Open search again.
6. Search for a page, and open the page.
7. Notice that the page opens, but also the searchbar with the search phrase, "t" that was typed in step 2 is open. 


The above error is fixed by making sure the navigationResult is cleared from `savedStateHandle` after being read.